olive/passes/pytorch/lora.py

```diff
@@ -214,20 +215,62 @@ def validate_search_point(
-                logger.info(
-                    "bfloat16 is not supported by onnxruntime-training yet. Please use a different torch_dtype."
-                )
-                return False
-            if search_point.get("training_args", {}).get("gradient_checkpointing"):
-                logger.info(
-                    "gradient_checkpointing is not supported by onnxruntime-training. Please set gradient_checkpointing"
-                    " to False."
-                )
-                return False
+        if search_point.get("use_ort_trainer") and search_point.get("training_args", {}).get("gradient_checkpointing"):
+            logger.info(
+                "gradient_checkpointing is not supported by onnxruntime-training. Please set gradient_checkpointing"
+                " to False."
+            )
+            return False
         return True
 
+    @classmethod
+    def check_dependencies(cls, config: ConfigBase, is_qlora: bool = False):
+        """Check dependencies for the pass."""
+        if config.use_ort_trainer:
+            # check for ort trainer dependencies
+            try:
+                from optimum.onnxruntime import ORTTrainer  # noqa: F401
+                from optimum.onnxruntime.utils import is_onnxruntime_training_available
+                from torch_ort import ORTModule  # noqa: F401
+
+                assert is_onnxruntime_training_available(), "onnxruntime-training is not available."
+            except (ImportError, AssertionError):
+                raise ImportError(
+                    "Please install `olive-ai[optimum,ort-training]` or `onnxruntime-training optimum torch-ort` to use"
+                    f" {cls.__name__} pass with use_ort_trainer=True."
+                ) from None
+
+            # check if model uses bfloat16
+            uses_bf16 = cls.get_torch_dtype(config.torch_dtype) == torch.bfloat16
+            if is_qlora and config.compute_dtype:
+                # qlora compute dtype might be different from torch dtype
+                uses_bf16 |= cls.get_torch_dtype(config.compute_dtype) == torch.bfloat16
+
+            from onnxruntime import __version__ as OrtVersion
+
+            # onnxruntime-training doesn't support bfloat16 fully until 1.17.0
+            if uses_bf16 and version.parse(OrtVersion) < version.parse("1.17.0"):
+                raise ImportError(
+                    f"Please install onnxruntime >= 1.17.0 to use {cls.__name__} with bfloat16 and"
+                    " use_ort_trainer=True."
+                )
+
+            # set the opset version to 16 if using bfloat16
+            if uses_bf16:
+                original_opset_version = os.environ.get("ORTMODULE_ONNX_OPSET_VERSION", None)
+                # will try to convert to int, if not possible, will set to -1
+                try:
+                    original_opset_version = int(original_opset_version)
+                except (ValueError, TypeError):
```

Comment:
a comment about when the two errors will be raised will be helpful to others to understand the background.

Reply:
done!

Reply:
it would be better if we add warning if user pass invalid env variables. If user doesn't provide the env variable, the log is needless.